homeassistant/components/velbus/config_flow.py

```diff
@@ -0,0 +1,71 @@
+"""Config flow for the Velbus platform."""
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_PORT, CONF_NAME
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.util import slugify
+
+from .const import DOMAIN
+
+
+@callback
+def velbus_entries(hass: HomeAssistant):
+    """Return connections for Velbus domain."""
+    return set((entry.data[CONF_PORT]) for
+               entry in hass.config_entries.async_entries(DOMAIN))
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class VelbusConfigFlow(config_entries.ConfigFlow):
+    """Handle a config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_PUSH
+
+    def __init__(self) -> None:
+        self._errors = {}
+
+    async def _create_device(self, name: str, prt: str):
+        return self.async_create_entry(
+            title=name,
+            data={
+                CONF_PORT: prt
+            }
+        )
+
+    def _prt_in_configuration_exists(self, prt: str) -> bool:
+        """Return True if port exists in configuration."""
+        if prt in velbus_entries(self.hass):
+            return True
+        return False
+
+    async def async_step_user(self, user_input=None):
+        """Step when user intializes a integration"""
+        self._errors = {}
+        if user_input is not None:
+            name = slugify(user_input[CONF_NAME])
+            prt = user_input.get(CONF_PORT)
+            # name must be unique
+            if not self._prt_in_configuration_exists(prt):
+                return await self._create_device(name, prt)
+            self._errors[CONF_NAME] = 'port_exists'
```

Comment:
If we use a key for the error that matches the form field item key, the error message will be shown next to that option field.

I suggest using `CONF_PORT` here.